File: next/common/jsbridge/signer/signer.go
package signer

// HasSigner returns true if window.nostr (NIP-07) is available.
func HasSigner() bool { panic("jsbridge") }

// GetPublicKey requests the public key from the browser extension.
// Calls fn with hex pubkey on success, empty string on failure.
func GetPublicKey(fn func(string)) { panic("jsbridge") }

// SignEvent signs a serialized nostr event via NIP-07.
// Calls fn with the signed event JSON on success, empty string on failure.
func SignEvent(eventJSON string, fn func(string)) { panic("jsbridge") }

// Nip04Decrypt decrypts NIP-04 ciphertext via the browser extension.
func Nip04Decrypt(peerPubkey, ciphertext string, fn func(string)) { panic("jsbridge") }

// Nip04Encrypt encrypts plaintext via NIP-04 through the browser extension.
func Nip04Encrypt(peerPubkey, plaintext string, fn func(string)) { panic("jsbridge") }

// Nip44Decrypt decrypts NIP-44 ciphertext via the browser extension.
func Nip44Decrypt(peerPubkey, ciphertext string, fn func(string)) { panic("jsbridge") }

// Nip44Encrypt encrypts plaintext via NIP-44 through the browser extension.
func Nip44Encrypt(peerPubkey, plaintext string, fn func(string)) { panic("jsbridge") }

File: app/smesh3/$runtime/signer.mjs
// TinyJS Runtime — NIP-07 Browser Signer Bridge

export function HasSigner() {
  return typeof window !== 'undefined' && !!window.nostr;
}

export function GetPublicKey(fn) {
  if (!window.nostr) { fn(''); return; }
  window.nostr.getPublicKey()
    .then(pubkey => fn(pubkey))
    .catch(() => fn(''));
}

export function SignEvent(eventJSON, fn) {
  if (!window.nostr) { console.error('signer: window.nostr not available'); fn(''); return; }
  try {
    const ev = JSON.parse(eventJSON);
    window.nostr.signEvent(ev)
      .then(signed => fn(JSON.stringify(signed)))
      .catch(e => { console.error('signer: signEvent rejected:', e); fn(''); });
  } catch(e) { console.error('signer: signEvent exception:', e); fn(''); }
}

export function Nip04Decrypt(peerPubkey, ciphertext, fn) {
  if (!window.nostr || !window.nostr.nip04) { fn(''); return; }
  window.nostr.nip04.decrypt(peerPubkey, ciphertext)
    .then(plain => fn(plain))
    .catch(() => fn(''));
}

export function Nip04Encrypt(peerPubkey, plaintext, fn) {
  if (!window.nostr || !window.nostr.nip04) { fn(''); return; }
  window.nostr.nip04.encrypt(peerPubkey, plaintext)
    .then(ct => fn(ct))
    .catch(() => fn(''));
}

export function Nip44Decrypt(peerPubkey, ciphertext, fn) {
  if (!window.nostr || !window.nostr.nip44) { fn(''); return; }
  window.nostr.nip44.decrypt(peerPubkey, ciphertext)
    .then(plain => fn(plain))
    .catch(() => fn(''));
}

export function Nip44Encrypt(peerPubkey, plaintext, fn) {
  if (!window.nostr || !window.nostr.nip44) { fn(''); return; }
  window.nostr.nip44.encrypt(peerPubkey, plaintext)
    .then(ct => fn(ct))
    .catch(() => fn(''));
}
